src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -59,7 +59,11 @@ def register_cli_argument(self, *args, **kwargs):
             register_cli_argument(*args, **kwargs)
         else:
             from azure.cli.core.commands.parameters import ignore_type
-            kwargs = {'arg_type': ignore_type}
+            if 'arg_type' in kwargs:
+                kwargs = {'arg_type': ignore_type}
+            else:
+                kwargs = {}
+                args = tuple([args[0], args[1], ignore_type])
```

Comment:
Guess i am not very familiar with ignored argument, but why we don't do the same ` kwargs = {'arg_type': ignore_type}` like the `if` branch above?